cmd/server/main.go

```diff
@@ -74,11 +73,21 @@ func main() {
 	r := mux.NewRouter()
 
 	// Setup rate limiter
-	limitInstance := limiter.New(memory.NewStore(), limiter.Rate{
-		Period: 1 * time.Minute,
-		Limit:  int64(config.RateLimit),
-	}, limiter.WithTrustForwardHeader(true))
-	r.Use(limithttp.NewMiddleware(limitInstance).Handler)
+	store, err := memorystore.New(&memorystore.Config{
+		Tokens:   config.RateLimit,
+		Interval: 1 * time.Minute,
+	})
+	if err != nil {
+		log.Fatalf("failed to create limiter: %v", err)
+	}
+	defer store.Stop()
+
+	httplimiter, err := httplimit.NewMiddleware(store, httplimit.IPKeyFunc("X-Forwarded-For"))
```

Comment:
I think this is where, for logged in actions, we want user based limiting, can be a followup.